lib/ansible/modules/tacp/tacp_info.py

```diff
@@ -0,0 +1,220 @@
+    # we primarily care about changed and state
+    # change is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        args=[]
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    if module.check_mode:
+        module.exit_json(**result)
+
+    def fail_with_reason(reason):
+        result['msg'] = reason
+        module.fail_json(**result)
+
+    # Return the inputs for debugging purposes
+    result['args'] = module.params
+
+    # Define configuration
+    configuration = tacp.Configuration()
+    configuration.host = "https://manage.cp.lenovo.com"
+    configuration.api_key_prefix['Authorization'] = 'Bearer'
+    configuration.api_key['Authorization'] = module.params['api_key']
+    api_client = tacp.ApiClient(configuration)
+
+    resource_dict = {"application": tacp_utils.ApplicationResource,
+                     "application_group": tacp_utils.ApplicationGroupResource,
+                     "category": tacp_utils.CategoryResource,
+                     "datacenter": tacp_utils.DatacenterResource,
+                     "firewall_profile": tacp_utils.FirewallProfileResource,
+                     "instance": tacp_utils.ApplicationResource,
+                     "marketplace_template": tacp_utils.MarketplaceTemplateResource,
+                     "migration_zone": tacp_utils.MigrationZoneResource,
+                     "site": tacp_utils.SiteResource,
+                     "storage_pool": tacp_utils.StoragePoolResource,
+                     "tag": tacp_utils.TagResource,
+                     "template": tacp_utils.TemplateResource,
+                     "user": tacp_utils.UserResource,
+                     "vlan": tacp_utils.VlanResource,
+                     "vnet": tacp_utils.VnetResource
+                     }
+
+    resource = resource_dict[module.params['resource']](api_client)
+
+    items = resource.filter(as_dict=True)
+
+    if module.params['resource'] == 'migration_zone':
+        application_resource = tacp_utils.ApplicationResource(
+            api_client)
+        category_resource = tacp_utils.CategoryResource(
+            api_client)
+        datacenter_resource = tacp_utils.DatacenterResource(
+            api_client)
+        vnet_resource = tacp_utils.VnetResource(api_client)
+        vlan_resource = tacp_utils.VlanResource(api_client)
```

Comment:
It looks like `vent_resource` and `vlan_resource` are defined here but then used in the `elif ... == 'datacenter'` block.

Reply:
fixed :)